src/main/java/Duke.java

```diff
@@ -1,10 +1,259 @@
+     */
+    private static String deadlineCommand(String args) {
+        if (!args.equals("")) {
+            String[] parsedArgs = args.trim().split(DEADLINE_SEPARATOR, 2);
+            if (parsedArgs.length == 2) {
+                final String taskDescription = parsedArgs[0].trim();
+                final String taskDeadline = parsedArgs[1].trim();
+                Deadline newDeadline = new Deadline(taskDescription, taskDeadline);
+                return addTask(newDeadline, MESSAGE_DEADLINE_ADDED);
+            }
+            return invalidCommandError(COMMAND_DEADLINE, USAGE_DEADLINE);
+        }
+        return invalidCommandError(COMMAND_DEADLINE, USAGE_DEADLINE);
+    }
+
+    /**
+     * Adds a new event task. Parsing of the event separator occurs in the function body.
+     */
+    private static String eventCommand(String args) {
+        if (!args.equals("")) {
+            String[] parsedArgs = args.trim().split(EVENT_SEPARATOR, 2);
+            if (parsedArgs.length == 2) {
+                final String eventDescription = parsedArgs[0].trim();
+                final String eventTime = parsedArgs[1].trim();
+                Event newEvent = new Event(eventDescription, eventTime);
+                return addTask(newEvent, MESSAGE_EVENT_ADDED);
+            }
+            return invalidCommandError(COMMAND_EVENT, USAGE_EVENT);
+        }
+        return invalidCommandError(COMMAND_EVENT, USAGE_EVENT);
+    }
+
+    /**
+     * Marks the specified task as completed.
+     */
+    private static String markCommand(String args) {
+        try {
+            int index = Integer.parseInt(args);
+            if (index > 0 && index <= numSavedTasks) {
+                if (savedTasks[index - 1].getIsDone()) {
+                    return MESSAGE_TASK_ALREADY_MARKED;
+                } else {
+                    savedTasks[index - 1].setIsDone(true);
+                    return String.format(MESSAGE_TASK_MARKED, index, savedTasks[index-1]);
+                }
+            } else {
+                return MESSAGE_NO_SUCH_INDEX;
+            }
+        } catch (Exception e) {
+            return invalidCommandError(COMMAND_MARK, USAGE_MARK);
+        }
+    }
+    /**
+     * Marks the specified task as incomplete.
+     */
+    private static String unmarkCommand(String args) {
+        try {
+            int index = Integer.parseInt(args);
+            if (index > 0 && index <= numSavedTasks) {
+                if (savedTasks[index - 1].getIsDone()) {
+                    savedTasks[index - 1].setIsDone(false);
+                    return String.format(MESSAGE_TASK_UNMARKED, index, savedTasks[index-1]);
+
+                } else {
+                    return MESSAGE_TASK_ALREADY_UNMARKED;
+                }
+            } else {
+                return MESSAGE_NO_SUCH_INDEX;
+            }
+        } catch (Exception e) {
+            return invalidCommandError(COMMAND_UNMARK, USAGE_UNMARK);
```

Comment:
This part is a little confusing and almost like an arrowhead code. I am not very sure how this part works but if there is a way to simplify the code and use less if else to make it easier to read then it will be great:)